Modules/Sources/Yosemite/Tools/POS/BatchedRequestLoader.swift

```diff
@@ -46,6 +83,40 @@ final class BatchedRequestLoader {
 
         return allItems
     }
+
+    private static func fetchPageWithRetry<T>(
+        pageNumber: Int,
+        maxRetries: Int,
+        retryDelay: TimeInterval,
+        errorEvaluator: RetryErrorEvaluator,
+        makeRequest: @escaping (Int) async throws -> PagedItems<T>
+    ) async throws -> PagedItems<T> {
+        var lastError: Error?
+
+        for attempt in 0..<maxRetries {
+            do {
+                return try await makeRequest(pageNumber)
+            } catch {
+                lastError = error
+
+                if !errorEvaluator.shouldRetry(error) {
+                    DDLogError("⛔️ Page \(pageNumber) failed with non-retryable error: \(error)")
+                    throw error
+                }
+
+                // Logs and retries with exponential backoff.
+                if attempt < maxRetries - 1 {
+                    let delay = retryDelay * pow(2.0, Double(attempt))
+                    DDLogWarn("⚠️ Page \(pageNumber) failed (attempt \(attempt + 1)/\(maxRetries)): \(error). Retrying in \(delay)s...")
+                    try await Task.sleep(nanoseconds: UInt64(delay * 1_000_000_000))
```

Comment:
```suggestion
                    try await Task.sleep(nanoseconds: UInt64(delay * Double(NSEC_PER_SEC))
```

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/16197/commits/a70aa257ea06a395070ac48c9c1c9c81f2196827.